pegasus/helpers/section_api_helpers.rb

```diff
@@ -21,23 +21,29 @@ def self.fetch_user_students(user_id)
   def self.create(params)
     name = !params[:name].to_s.empty? ? params[:name].to_s : 'New Student'
     gender = valid_gender?(params[:gender]) ? params[:gender] : nil
+    provider = valid_provider?(params[:provider]) ? params[:provider] : 'sponsored'
     birthday = age_to_birthday(params[:age]) ?
       age_to_birthday(params[:age]) : params[:birthday]
 
     created_at = DateTime.now
 
-    row = Dashboard.db[:users].insert(
+    data =
       {
         name: name,
         user_type: 'student',
-        provider: 'sponsored',
+        provider: provider,
         gender: gender,
         birthday: birthday,
         created_at: created_at,
         updated_at: created_at,
         username: UserHelpers.generate_username(Dashboard.db[:users], name)
-      }.merge(random_secrets)
-    )
+      }
+    if provider == 'sponsored'
+      data.merge!(random_secrets)
```

Comment:
In addition, there were thirty-three users without secret words, all of whom were also missing a secret picture. These users were "fixed" with the following snippet run from `dashboard-console`.
```
user_ids.each do |user_id|
  u = User.with_deleted.find_by_id(user_id)
  u.generate_secret_picture
  u.generate_secret_words
  u.save(validate: false)
end
```